include/engine/hdf5/HDF5WriterP.h

```diff
@@ -0,0 +1,224 @@
+   * @param name unique name given to the engine
+   * @param accessMode
+   * @param mpiComm
+   * @param method
+   * @param debugMode
+   */
+
+  HDF5Writer(ADIOS &adios, const std::string name, const std::string accessMode,
+             MPI_Comm mpiComm, const Method &method,
+             const bool debugMode = false, const unsigned int cores = 1);
+
+  ~HDF5Writer();
+
+  void Write(Variable<char> &variable, const char *values);
+  void Write(Variable<unsigned char> &variable, const unsigned char *values);
+  void Write(Variable<short> &variable, const short *values);
+  void Write(Variable<unsigned short> &variable, const unsigned short *values);
+  void Write(Variable<int> &variable, const int *values);
+  void Write(Variable<unsigned int> &variable, const unsigned int *values);
+  void Write(Variable<long int> &variable, const long int *values);
+  void Write(Variable<unsigned long int> &variable,
+             const unsigned long int *values);
+  void Write(Variable<long long int> &variable, const long long int *values);
+  void Write(Variable<unsigned long long int> &variable,
+             const unsigned long long int *values);
+  void Write(Variable<float> &variable, const float *values);
+  void Write(Variable<double> &variable, const double *values);
+  void Write(Variable<long double> &variable, const long double *values);
+  void Write(Variable<std::complex<float>> &variable,
+             const std::complex<float> *values);
+  void Write(Variable<std::complex<double>> &variable,
+             const std::complex<double> *values);
+  void Write(Variable<std::complex<long double>> &variable,
+             const std::complex<long double> *values);
+
+  void Write(const std::string variableName, const char *values);
+  void Write(const std::string variableName, const unsigned char *values);
+  void Write(const std::string variableName, const short *values);
+  void Write(const std::string variableName, const unsigned short *values);
+  void Write(const std::string variableName, const int *values);
+  void Write(const std::string variableName, const unsigned int *values);
+  void Write(const std::string variableName, const long int *values);
+  void Write(const std::string variableName, const unsigned long int *values);
+  void Write(const std::string variableName, const long long int *values);
+  void Write(const std::string variableName,
+             const unsigned long long int *values);
+  void Write(const std::string variableName, const float *values);
+  void Write(const std::string variableName, const double *values);
+  void Write(const std::string variableName, const long double *values);
+  void Write(const std::string variableName, const std::complex<float> *values);
+  void Write(const std::string variableName,
+             const std::complex<double> *values);
+  void Write(const std::string variableName,
+             const std::complex<long double> *values);
+
+  void Close(const int transportIndex = -1);
+  /*
```

Comment:
what do you mean by "Dead code, now in the cpp"? I see the BPFileWriter.h
still has it.